experimental/filesystem-backends.mdx

```diff
@@ -0,0 +1,332 @@
+The configuration contains fields for addressing the repository, configuring the target reference as well as adding authentication credentials.
+
+Once a target repository and reference is configured, Flipt will poll the source repository on a periodic cadence.
+This cadence is also configurable and defaults to 30 seconds.
+
+Flipt will follow the configured reference (e.g. branch name) and keep up to date with new changes.
+
+<Tabs>
+  <Tab title="Environment Variable">
+
+    ```bash
+    FLIPT_EXPERIMENTAL_FILESYSTEM_STORAGE_ENABLED=true
+    FLIPT_STORAGE_TYPE="git"
+    FLIPT_STORAGE_GIT_REPOSITORY="https://github.com/predictab.le/config.git"
+    FLIPT_STORAGE_GIT_REF="main"
+    FLIPT_STORAGE_GIT_POLL_INTERVAL="30s"
+    # for private repository access
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_USERNAME=...
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_PASSWORD=...
+    ```
+
+  </Tab>
+  <Tab title="Configuration Yaml">
+
+    ```yaml
+    experimental:
+      filesystem_storage:
+        enabled: true
+
+    storage:
+      type: git
+      git:
+        repository: "https://github.com/predictab.le/config.git"
+        ref: "main"
+        poll_interval: "30s"
+        authentication:
+          basic:
+            username: ...
+            password: ...
+          token:
+            access_token: ...
+    ```
+
+  </Tab>
+</Tabs>
+
+#### Authentication
+
+The `git` type backend supports both `basic` and `token` based authentication schemes.
+These sections are mutually exclusive. Flipt will fail to initialize with a validation error if you attempt to configure both.
+
+<Note>
+  GitHub PAT (Personal Access Tokens) should be supplied using `basic`
+  authentication as the password. Use your GitHub username in the `username`
+  field.
+</Note>
+
+## Flag State Configuration
+
+Each of Flipts filesystem backends expects you to represent your feature flag configuration via a set of yaml files.
+These files declaratively define what flags, segments, variants etc. exist and in what configuration.
+
+### Locating Flag State
+
+Flipts filesystem backends allow you to define feature flags alongside other configuration in a shared directory or repository.
+In order to identify which files are intended for Flipt, it uses a naming scheme to index which are flag state files.
+By default, Flipt will look for the following filename patterns when identifying files to attempt to parse as Flipt state:
+
+- `**/features.yaml`
+- `**/features.yml`
+- `**/*.features.yaml`
+- `**/*.features.yml`
+
+Meaning, any file named `features.yaml`, `features.yml` or with either extension `.features.yaml` or `.features.yml` is considered.
+This is recursively from the root of your target directory or repository.
+
+If this naming convention does not work for you, for whatever reason, it can be overriden by creating a file named `.flipt.yml` in the root of your target directory tree.
```

Comment:
```suggestion
If this naming convention does not work for you, for whatever reason, it can be overridden by creating a file named `.flipt.yml` in the root of your target directory tree.
```